src/js/components/utils.tsx

```diff
@@ -36,6 +41,105 @@ import {
 import routes from '@/utils/routes';
 import githubIcon from '#/github.svg';
 
+interface Crumb {
+  name: string;
+  url?: string;
+}
+
+export const DetailPageLayout = ({
+  title,
+  description,
+  repoUrl,
+  breadcrumb,
+  onRenderHeaderActions,
+  sidebar,
+  children,
+}: {
+  title: string;
+  description?: string;
+  repoUrl: string;
+  breadcrumb: Crumb[];
+  onRenderHeaderActions?: () => JSX.Element;
+  sidebar?: ReactNode;
+  children?: ReactNode;
+}) => {
+  const descriptionHasTitle =
+    description &&
+    (description.startsWith('<h1>') || description.startsWith('<h2>'));
+
+  return (
+    <>
+      <PageHeader
+        className="page-header slds-p-around_x-large"
+        title={title}
+        info={<RepoLink url={repoUrl} shortenGithub />}
+        onRenderControls={onRenderHeaderActions}
+      />
+      <div
+        className="slds-p-horizontal_x-large
+          slds-p-top_x-small
+          ms-breadcrumb"
+      >
+        <BreadCrumb
+          trail={[
+            <Link to={routes.home()} key="home">
+              {i18n.t('Home')}
+            </Link>,
+          ].concat(
```

Comment:
Does `concat` return the concatenated arrays? Different than the interface I expected, but makes sense.

Reply:
Yep.

> This method does not change the existing arrays, but instead returns a new array.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/concat